agent/tool-scripts/base-tool

```diff
@@ -595,6 +595,15 @@ install)
 		check_required_rpm ${tool_package_name} ${tool_package_ver}
 		rc=${?}
 		;;
+	pprof)
+		check_required_rpm ${tool_package_name} ${tool_package_ver}
+		rc=${?}
+		if [[ ${rc} == 0 ]]; then
+            # For pprof, we also need ansible installed.
+		    check_required_rpm "ansible"
+			rc=${?}
+		fi
+		;;
```

Comment:
Since `pprof` is in the list at line 594, does that mean that it will execute that case _instead of_ the case at line 598 or _in addition to_ it?

Given that lines 595-597 are duplicated among the lines that you've added, I would be inclined to combine the cases -- if the `if` test checks both `${rc}` and `${tool}`, then you don't need a separate case and you can avoid the other duplication.

Also, line 602 appears to be indented with spaces when the surrounding lines are indented using hard-tabs.

Reply:
Okay, the first and last items above I have addressed in a simple code-review commit update, but the middle item actually hit a bit larger bug in the `base-tool` script where some tools were not being properly checked.  So I made that an entirely separate commit.

Reply:
Do you intend to squash the three commits together when you merge?  (I hope you do....)

If not, then please remove the `pprof` reference from line 594 in the same commit that you add the new case for it.

Reply:
> Do you intend to squash the three commits together when you merge? (I hope you do....)

The first two, yes, but I'd like to keep the second separate since it is a more radical change, a bug fix on its own, and refactoring that is not required to make the intended behavior work.

> 
> If not, then please remove the `pprof` reference from line 594 in the same commit that you add the new case for it.

I believe by squashing the first two commits together, this concern has been addressed.